### DISCUSSION TITLE
ADBC Adapter

### DISCUSSION THREAD

**Discussion by TylerHillery** [tconbeer/harlequin#446] [Ideas (Feature Requests)]
# Feature Overview
Add the ability to connect via ADBC (similar to the ODBC adapter but better because it uses arrow 😁). I am going to start working on this adapter here [harlequin-adbc](https://github.com/TylerHillery/harlequin-adbc).

Below are the packages I plan on leveraging to build this:
- Required
  - [ADBC Driver Manager](https://arrow.apache.org/adbc/current/python/driver_manager.html)

- Optional
  - [Flight SQL Driver](https://arrow.apache.org/adbc/current/driver/flight_sql.html)
  - [Postgres Driver](https://arrow.apache.org/adbc/current/driver/postgresql.html)
  - [Snowflake Driver](https://arrow.apache.org/adbc/current/driver/snowflake.html)
  - [SQLite Driver](https://arrow.apache.org/adbc/current/driver/sqlite.html)
  
Let me know if anyone has any input on the design or implementation of this adapter.

**Comment by alexmalins** [comment]
Hey Tyler 👋 This is cool.

I'm a newbie with Arrow/ADBC, but you may find some of the code in the [Databricks adapter](https://github.com/alexmalins/harlequin-databricks/blob/main/src/harlequin_databricks/adapter.py) useful to refer to.

Databricks's Python-SQL connector has methods that return Arrow objects natively, so I used those for executing queries and for building the catalog. Or may be you can point out ways it could have been done better / more efficiently. 🙈

**Reply by TylerHillery** [reply]
Hey Alex! This is actually really helpful because I was wondering if there were ways I would be able to return Arrow objects natively. I will definitely take a look at the Databricks connector to see how you're doing it. Thanks!

**Comment by tconbeer** [comment]
Closing since this has been shipped!
